clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -743,8 +750,11 @@ public void handleResponse(AbstractResponse response) {
                     lookupCoordinator(FindCoordinatorRequest.CoordinatorType.TRANSACTION, transactionalId);
                     reenqueue();
                     return;
-                } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.CONCURRENT_TRANSACTIONS
-                        || error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
+                } else if (error == Errors.CONCURRENT_TRANSACTIONS) {
+                    retryBackoffMs = 5;
```

Comment:
Actually a backoff of 10ms is significantly faster than 15ms on my local test (where network is not a bottleneck). I think a lower number is better. With a 15ms delay, transactions are 30% slower than a regular producer. With a 10ms delay, transactions are 20% slower..